### PR TITLE
Implement channel close resolving

### PR DESCRIPTION
**Pull request by dleutenegger**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1520,13 +1535,12 @@ impl LightningNode {
...
+        let rate = self.get_exchange_rate();
         Ok(SweepInfo {
             address,
             onchain_fee_rate,
-            onchain_fee_sat: res
-                .sweep_tx_fee_sat
-                .as_sats()
-                .to_amount_up(&self.get_exchange_rate()),
+            onchain_fee_sat: res.sweep_tx_fee_sat.as_sats().to_amount_up(&rate),
+            amount: res.sweep_tx_fee_sat.as_sats().to_amount_up(&rate),
```
[the remark this answers is not in the document]
Comment (dleutenegger) [reply src/lib.rs]: It is, I've addressed it in f2aca5e

[the remark this answers is not in the document]
Comment (danielgranhao) [reply src/lib.rs]: I think it's worth documenting the field to make it clear it's the value that will be sent. Not the amount available from which onchain fees are then deducted.

File: src/lib.rs [hunk 2]
```diff
@@ -255,6 +258,18 @@ pub struct LightningNode {
     environment: Environment,
 }
 
+/// Contains the fee information for the options to resolve on-chain funds from channel closes.
+pub struct ChannelCloseResolvingFees {
+    /// Fees to swap the funds back to lightning using [`LightningNode::swap_onchain_to_lightning`]
+    /// Only available if enough funds are there to swap.
+    pub swap_fees: Option<SwapToLightningFees>,
+    /// Estimate of the fees for sending the funds on-chain using [`LightningNode::sweep`].
```
Comment (danielgranhao) [inline src/lib.rs:266]:
```suggestion
    /// Estimate of the fees for sending the funds on-chain using [`LightningNode::sweep`].
    /// The exact fees will be known when calling [`LightningNode::prepare_sweep`].
```

File: examples/node/cli.rs [hunk 3]
```diff
@@ -1176,3 +1196,45 @@ fn calculate_lightning_payout_fee(
...
+    let resolving_fees = node.get_channel_close_resolving_fees()?;
+
+    println!(
+        "On-chain fees: {}",
+        amount_to_string(resolving_fees.onchain_fees)
+    );
+
+    match resolving_fees.swap_fees {
+        Some(f) => {
+            println!("Swap fees: {}", amount_to_string(f.total_fees));
```
[the remark this answers is not in the document]
Comment (danielgranhao) [reply examples/node/cli.rs:1210]: You seem to have applied my suggestion in the wrong place :)

**Review by danielgranhao — APPROVED** [review]

**Pull request merged**